store/store.go

```diff
@@ -40,6 +43,7 @@ type ProjectJobSpecRepository interface {
 type ProjectRepository interface {
 	Save(context.Context, models.ProjectSpec) error
 	GetByName(context.Context, string) (models.ProjectSpec, error)
+	GetByID(context.Context, uuid.UUID) (models.ProjectSpec, error)
```

Comment:
Please create a new model in Project for the `ProjectID` and use it in here.

Reply:
deleted `GetByID` from project repository as it is only being used inside `project_repository_test`. to verify the test, i have replaced `GetByID` with `GetByName`.